tests/functional/test_report_only_no_errors.py

```diff
@@ -0,0 +1,319 @@
+    pr = json.loads(r.text)
+
+    r = requests.get(
+        f'https://api.github.com/repos/{secrets.test_repo}/actions/runs', headers=headers)
+    runs = json.loads(r.text)
+
+    for run in runs['workflow_runs']:
+        if run['head_sha'] == pr['head']['sha']:
+            return run['id']
+    else:
+        pytest.fail("Workflow for the submitted PR did not run.")
+
+
+@retry(stop_max_delay=60_000*2)
+def get_run_result(headers, secrets, run_id):
+    r = requests.get(
+        f'https://api.github.com/repos/{secrets.test_repo}/actions/runs/{run_id}', headers=headers)
+    run = json.loads(r.text)
+
+    if run['conclusion'] is None:
+        pytest.fail("Workflow conclusion is null.")
+
+    return run['conclusion']
+
+
+@pytest.fixture
+def secrets():
+    @dataclass
+    class Secret:
+        test_repo: str
+        fork_repo: str
+        cluster_token: str
+        bot_token: str
+        bot_name: str
+        pr_base_branch: str
+        fork_branch: str
+
+        pr_number: int = -1
+        vendor_type: str = ''
+        vendor: str = ''
+        owners_file_content: str = """\
+chart:
+  name: ${chart_name}
+  shortDescription: Test chart for testing chart submission workflows.
+publicPgpKey: null
+users:
+- githubUsername: ${bot_name}
+vendor:
+  label: ${vendor}
+  name: ${vendor}
+"""
+        test_report: str = 'tests/data/report-no-errors.yaml'
+        chart_name, chart_version = get_name_and_version_from_report(
+            test_report)
+
+    test_repo = os.environ.get("TEST_REPO")
+    if not test_repo:
+        raise Exception("TEST_REPO environment variable not defined")
+    fork_repo = os.environ.get("FORK_REPO")
+    if not fork_repo:
+        raise Exception("FORK_REPO environment variable not defined")
+    bot_name = fork_repo.split("/")[0]
+    cluster_token = os.environ.get("CLUSTER_TOKEN")
+    if not cluster_token:
+        raise Exception("CLUSTER_TOKEN environment variable not defined")
+    bot_token = os.environ.get("BOT_TOKEN")
+    if not bot_token:
+        bot_name = "github-actions[bot]"
+        bot_token = os.environ.get("GITHUB_TOKEN")
+        if not bot_token:
+            raise Exception("BOT_TOKEN environment variable not defined")
+
+    pr_base_branch = 'test-charts'
+    fork_branch = pr_base_branch + '-pr'
+    test_repo = str(base64.b64decode(test_repo), encoding="utf-8")
+    fork_repo = str(base64.b64decode(fork_repo), encoding="utf-8")
+    yield Secret(test_repo, fork_repo, cluster_token, bot_token, bot_name, pr_base_branch, fork_branch)
+
+    # Teardown step to cleanup branches
+    headers = {'Accept': 'application/vnd.github.v3+json',
+               'Authorization': f'Bearer {bot_token}'}
+    requests.delete(
+        f'https://api.github.com/repos/{test_repo}/git/refs/heads/{pr_base_branch}-gh-pages', headers=headers)
+    requests.delete(
+        f'https://api.github.com/repos/{fork_repo}/git/refs/heads/{fork_branch}', headers=headers)
```

Comment:
need to also delete releases and tags.

Reply:
since this is report only so there wouldn't be a release.. only index updates